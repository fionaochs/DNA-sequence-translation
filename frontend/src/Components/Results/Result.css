h3 {
    font-size: 1.5rem;

}

.sequenceResult {
    display: flex;
    justify-content: center;
    flex-direction: column;
    align-items: center;
}
