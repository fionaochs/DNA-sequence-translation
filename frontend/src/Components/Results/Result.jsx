import React from 'react';
import styles from './Result.css';

const Result = ({DNASequence, proteinName, organism, proteinLocation}) => {

    const sequenceNotFound = proteinName === undefined;

    return (
        <div className={styles.sequenceResult}>
            {sequenceNotFound ? (
                <div>
                    <h3>DNA sequence <span>{DNASequence}</span></h3>
                    <h3>did not match any organisms on file</h3>
                </div>
            ) : (
                <div>
                    <h3 className={DNASequence}>Found DNA sequence <span>{DNASequence}</span></h3>
                    <h3>at location <span>{proteinLocation}</span></h3>
                    <h3>in protein <span>{proteinName}</span></h3>
                    <h3>in organism <span>{organism}</span></h3>
                </div>
            )
            }
        </div>
    );
};

export default Result;