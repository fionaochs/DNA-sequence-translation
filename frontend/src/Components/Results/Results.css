.results {
    display: grid;
    grid-template-columns: repeat(4, 1fr);
    /*justify-content: center;*/
    /*flex-direction: row;*/
    /*align-items: center;*/
}

h2 {
    margin-top: 2rem;
    font-size: 2rem;
}

ul {
    border-radius: 1rem;
    border-style: solid;
    border-color: aliceblue;
    padding-right: 1rem;
    background-color: #1A8CA9;
}