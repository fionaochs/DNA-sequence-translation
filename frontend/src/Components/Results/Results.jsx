import React, {useEffect, useState} from 'react';
import Result from "./Result";
import {FetchResults, FetchTestResults} from "../../services/requests";
import {useLocalStorage} from "../../useLocalStorage";

const Results = () => {
    // const [proteinName, setProteinName] = useState([]);
    // const [results, setResults] = useState([]);
    // const [results, setResults] = useState([{'proteinName': 'protein 1', 'proteinLocation': 'protein 1 location'}, {'proteinName': 'protein 2', 'proteinLocation': 'protein 2 location'}]);

    // const [proteinName, setProteinName] = useState(() => {
    //     // getting stored value
    //     const saved = localStorage.getItem("proteinName");
    //     const initialValue = JSON.parse(saved);
    //     return initialValue || "";
    // });
    const [results, setResults] = useLocalStorage("results", []);

    // useEffect(() => {
    //     FetchTestResults()
    //         // .then(res => setResults(res));
    // }, []);
    // console.log(shoes)

    const resultsList = results.map(result =>
        <ul key={result.id}>
            <li key={result.name}>
                <Result {...result} />
            </li>
        </ul>
    );
    // const { results } = FetchTestResults();
    return (
        <>
            <h2>Results</h2>
            {/*<Result list={...results}/>*/}
            {resultsList}
        </>
    );
};
export default Results;