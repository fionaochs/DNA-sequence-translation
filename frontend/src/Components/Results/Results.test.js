import React from 'react';
import {render, screen, fireEvent, waitFor, cleanup} from '@testing-library/react';
import Results from './Results';

describe('Rating component', () => {
    afterEach(() => cleanup());
    it('renders Results list', () => {
        const {asFragment} = render(<Results/>);
        expect(asFragment()).toMatchSnapshot();
    });
});