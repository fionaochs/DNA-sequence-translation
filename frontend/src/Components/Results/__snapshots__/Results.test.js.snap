// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating component renders Results list 1`] = `
<DocumentFragment>
  <div>
    <div />
    <div />
  </div>
</DocumentFragment>
`;
